atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -0,0 +1,254 @@
+
+        if not clusters:
+            raise RuntimeError('No clusters found for platform {}!'
+                               .format(platform))
+
+        selected = min(clusters, key=lambda c: c.load)
+        self.log.info('platform %s will use cluster %s',
+                      platform, selected.cluster.name)
+        return selected
+
+    def get_release(self):
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow).labels
+        return get_preferred_label(labels, 'release')
+
+    def get_worker_build_kwargs(self, release, platform):
+        build_kwargs = deepcopy(self.build_kwargs)
+
+        build_kwargs.pop('architecture', None)
+
+        build_kwargs['release'] = release
+        build_kwargs['platform'] = platform
+
+        return build_kwargs
+
+    def do_worker_build(self, release, cluster_info):
+        build = None
+        try:
+            kwargs = self.get_worker_build_kwargs(release, cluster_info.platform)
+            build = cluster_info.osbs.create_worker_build(**kwargs)
+        except Exception:
+            self.log.exception('%s - failed to create worker build',
+                               cluster_info.platform)
+
+        build_info = WorkerBuildInfo(build=build, cluster_info=cluster_info)
+        self.worker_builds.append(build_info)
+
+        if build_info.build:
+            try:
+                self.log.info('%s - created build %s', cluster_info.platform,
+                              build_info.name)
+                build_info.watch_logs(self.log)
+                build_info.wait_to_finish()
+            except Exception as e:
+                build_info.monitor_exception = e
+                self.log.exception('%s - failed to monitor worker build',
+                                   cluster_info.platform)
+
+    def run(self):
+        release = self.get_release()
+        platforms = self.get_platforms()
+
+        thread_pool = ThreadPool(len(platforms))
+        result = thread_pool.map_async(
+            lambda cluster_info: self.do_worker_build(release, cluster_info),
+            [self.choose_cluster(platform) for platform in platforms]
+        )
+
+        try:
+            while not result.ready():
+                # The wait call is a blocking call which prevents signals
+                # from being processed. Wait for short intervals instead
+                # of a single long interval, so build cancellation can
+                # be handled virtually immediately.
+                result.wait(1)
+        # Always clean up worker builds on any error to avoid
+        # runaway worker builds (includes orchestrator build cancellation)
+        except Exception:
+            self.log.info('build cancelled, cancelling worker builds')
+            if self.worker_builds:
+                ThreadPool(len(self.worker_builds)).map(
+                    lambda bi: bi.cancel_build(), self.worker_builds)
+            while not result.ready():
```

Comment:
Is `result` the right thing at this point? We never assigned the result of the map() call to anything.

Since it's map() we're using anyway, I guess we don't need this polling loop in any case?